### PR TITLE
Remaining Bomb Bag and All-Night Mask

### PR DESCRIPTION
**Pull request by Eblo**
Done together because, again, they're directly related to each other.

- This bomb bag (normally the Big Bomb Bag) is a special case in the vanilla game. It can be obtained either by saving the Bomb Shop lady on the first night and buying it from the Bomb Shop after, or by not saving her and then buying it from the Curiosity Shop on the last day. These are mutually exclusive routes for reaching the same check.
  - The check is defined as existing in the Bomb Shop scene. I don't know if there currently exists a way to define multiple scenes for the same check. This will become a more pressing concern when entrance shuffling is on the table.
- As noted, there is a minuscule bug with buying the Curiosity Shop's special item and then trying to sell immediately afterward. It doesn't break anything, it requires a specific sequence of actions to trigger, and trying to fix it has proven troublesome. So I'm leaving it with the comment that it's known.
- ~~Seed generation started failing for `RC_BOMB_SHOP_ITEM_3`. It worked for me numerous times before it started failing. In theory, it should work. I commented out the problematic condition with a note.~~
- Perhaps overzealously I altered some dialog about these two items to instead describe the randomized replacements. In the case of the Bomb Shop, it lets the player know what the other item is before they actually see it.

### Build Artifacts
  - [2ship-linux.zip](https://nightly.link/garrettjoecox/2ship2harkinian/actions/artifacts/2370036183.zip)
  - [2ship-mac.zip](https://nightly.link/garrettjoecox/2ship2harkinian/actions/artifacts/2370036988.zip)
  - [2ship-windows.zip](https://nightly.link/garrettjoecox/2ship2harkinian/actions/artifacts/2370042272.zip)

### CODE REVIEW

File: mm/2s2h/Rando/Logic/Regions/Central.cpp [hunk 1]
```diff
@@ -51,8 +51,10 @@ static RegisterShipInitFunc initFunc([]() {
         .checks = {
             CHECK(RC_BOMB_SHOP_ITEM_1, CAN_AFFORD(RC_BOMB_SHOP_ITEM_1)),
             CHECK(RC_BOMB_SHOP_ITEM_2, CAN_AFFORD(RC_BOMB_SHOP_ITEM_2)),
-            CHECK(RC_BOMB_SHOP_ITEM_3, CAN_AFFORD(RC_BOMB_SHOP_ITEM_3)),
-            // TODO: Bigger bomb bag
+            // Upon saving the Bomb Shop lady, one item in the shop gets replaced with the other for the remainder of the cycle.
+            // The latter condition should be valid, but it started causing seed generation timeouts for this check.
+            CHECK(RC_BOMB_SHOP_ITEM_3, CAN_AFFORD(RC_BOMB_SHOP_ITEM_3) /*&& !CHECK_WEEKEVENTREG(WEEKEVENTREG_RECOVERED_STOLEN_BOMB_BAG)*/),
```
Comment (garrettjoecox) [inline mm/2s2h/Rando/Logic/Regions/Central.cpp]: Looks like you accidentally had an `!` here, remove that and you should be good

File: mm/2s2h/Rando/Logic/Regions/Central.cpp [hunk 2]
```diff
@@ -197,7 +202,8 @@ static RegisterShipInitFunc initFunc([]() {
     };
     Regions[RR_CURIOSITY_SHOP_FRONT] = RandoRegion{ .name = "Front", .sceneId = SCENE_AYASHIISHOP,
         .checks = {
-            // TODO : Add the shop checks
+            CHECK(RC_BOMB_SHOP_ITEM_OR_CURIOSITY_SHOP_ITEM, CAN_AFFORD(RC_BOMB_SHOP_ITEM_OR_CURIOSITY_SHOP_ITEM) && !CHECK_WEEKEVENTREG(WEEKEVENTREG_RECOVERED_STOLEN_BOMB_BAG)),
```
Comment (garrettjoecox) [inline mm/2s2h/Rando/Logic/Regions/Central.cpp]: I'm surprised this one is passing, but you should remove the unset flag check here, the logic doesn't remove flags unless the scene is no longer in logic currently.
```suggestion
            CHECK(RC_BOMB_SHOP_ITEM_OR_CURIOSITY_SHOP_ITEM, CAN_AFFORD(RC_BOMB_SHOP_ITEM_OR_CURIOSITY_SHOP_ITEM)),
```

**Comment by Eblo** [comment]
Removed the negated checks

**Comment by Eblo** [comment]
Pushed another minor bugfix. I forgot this actor can sell stolen items, which are not checks right now, so now the actor behavior file only runs for the two checks.

**Pull request merged**